src/hooks/useRequest.ts

```diff
@@ -0,0 +1,23 @@
+import { DocumentNode } from 'graphql';
+import { useRef, useMemo } from 'react';
+import { GraphQLRequest } from '../types';
+import { createRequest } from '../utils';
+
+/** Creates a request from a query and variables but preserves reference equality if the key isn't changing */
+export const useRequest = (
+  query: string | DocumentNode,
+  variables?: any
```

Comment:
How would you feel about using a generic here so we can pass it through from `useQuery` / `useSubscription`? Happy to make a quick PR if you think it's worthwhile.